kola/tests/rpmostree/deployments.go

```diff
@@ -0,0 +1,300 @@
+	}
+
+	// checksum should be new commit
+	if postRebootStatus.Deployments[0].Checksum != string(newCommit) {
+		c.Fatalf(`New deployment checksum is incorrect; expected %q, got %q`, newCommit, postRebootStatus.Deployments[0].Checksum)
+	}
+
+	// version should be new version string
+	if postRebootStatus.Deployments[0].Version != "kola-test-1.0" {
+		c.Fatalf(`New deployment version is incorrect; expected %q, got %q`, "kola-test-1.0", postRebootStatus.Deployments[0].Checksum)
+	}
+
+	// fix original origin file to point to proper ref
+	originalOriginFile := "/ostree/deploy/rhcos/deploy/" + originalCsum + ".0.origin"
+	newSedCmd := "sudo sed -i 's|^refspec=.*$|refspec=" + originalOrigin + "|' " + originalOriginFile
+	c.MustSSH(m, newSedCmd)
+
+	// rollback to original deployment
+	c.MustSSH(m, "sudo rpm-ostree rollback")
+
+	newRebootErr := m.Reboot()
+	if newRebootErr != nil {
+		c.Fatalf("Failed to reboot machine: %v", err)
+	}
+
+	rollbackStatus, err := getRpmOstreeStatusJSON(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// validate we are back to the original deployment by comparing the
+	// the two rpmOstreeDeployment structs
+	if !reflect.DeepEqual(originalStatus.Deployments[0], rollbackStatus.Deployments[0]) {
+		c.Fatalf(`Differences found in "rpm-ostree status"; original %v, current: %v`, originalStatus.Deployments[0], rollbackStatus.Deployments[0])
+	}
+
+	// cleanup our mess
+	cleanupErr := rpmOstreeCleanup(c, m)
+	if cleanupErr != nil {
+		c.Fatal(cleanupErr)
+	}
+}
+
+// rpmOstreeInstallUninstall verifies that we can install a package
+// and then uninstall it
+func rpmOstreeInstallUninstall(c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	originalStatus, err := getRpmOstreeStatusJSON(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	originalCsum := originalStatus.Deployments[0].Checksum
+
+	// install fpaste and reboot
+	c.MustSSH(m, "sudo rpm-ostree install fpaste")
+
+	installRebootErr := m.Reboot()
+	if installRebootErr != nil {
+		c.Fatalf("Failed to reboot machine: %v", installRebootErr)
+	}
+
+	postInstallStatus, err := getRpmOstreeStatusJSON(c, m)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	if len(postInstallStatus.Deployments) != 2 {
+		c.Fatalf(`Expected two deployments, found %d deployments`, len(postInstallStatus.Deployments))
+	}
+
+	// check the command is present, in the rpmdb, and usable
+	cmdOut := c.MustSSH(m, "command -v fpaste")
+	if string(cmdOut) != "/bin/fpaste" {
+		c.Fatalf(`fpaste binary in unexpected location. expectd %q, got %q`, "/bin/fpaste", string(cmdOut))
+	}
+
+	// e.g. fpaste-0.3.7.4.1-2.el7.noarch
+	rpmOut := c.MustSSH(m, "rpm -q fpaste")
+	rpmMatch := regexp.MustCompile("^fpaste.*noarch").MatchString(string(rpmOut))
+	if !rpmMatch {
+		c.Fatalf(`Output from "rpm -q" was unexpected: %q`, string(rpmOut))
+	}
+
+	// e.g. fpaste 0.3.7.4.1
+	verOut := c.MustSSH(m, "fpaste --version")
+	verMatch := regexp.MustCompile(`^fpaste \d+\.\d+\.\d+\.\d+\.\d+`).MatchString(string(verOut))
```

Comment:
Why test the versioning scheme?  Doesn't seem worth it to potentially have a fire drill if fpaste changes its versioning scheme.  Maybe just test `fpaste --version` exits successfully?